newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -227,6 +229,7 @@ export default class AuthenticatedUserProvider extends React.Component<
         console.error('Error while loading user limits:', error);
       }
     );
+    this._onBadgesChanged();
```

Comment:
should this function rather be called `_fetchUserBadges` and passed to the context via the function `onBadgesChanges = this._fetchUserBadges` ?

Reply:
Agree with this: 
- the function should be named after what it does: this._fetchUserBadges. 
- but it is passed to the context with a name indicating when it should be called (when badges are changed: onBadgesChanged).

These are two different responsibilities. onBadgesChanged is plugged into this._fetchUserBadges.